packages/myst-transforms/src/glossary.ts

```diff
@@ -1,13 +1,86 @@
 import type { Plugin } from 'unified';
 import type { Node } from 'myst-spec';
 import type { VFile } from 'vfile';
-import { selectAll } from 'unist-util-select';
+import { selectAll, select } from 'unist-util-select';
 import type { GenericParent } from 'myst-common';
-import { normalizeLabel, toText, fileError, RuleId } from 'myst-common';
+import { normalizeLabel, toText, fileError, fileWarn, RuleId } from 'myst-common';
+import type { DefinitionTerm, DefinitionDescription, DefinitionList } from 'myst-spec-ext';
 
-export function glossaryTransform<T extends Node | GenericParent>(mdast: T, file: VFile) {
+export type Options = {
+  upgradeLegacySyntax?: boolean;
+};
+
+/**
+ * Lift out definitionTerms from top-level `TERM\nDEFINITION` text nodes
+ */
+function maybeLiftLegacyDefinitionTerm(node: GenericParent): GenericParent | undefined {
```

Comment:
I know this pattern was confusing in the epigraphs PR.

It either identifiers `paragraph > text[value*=\n]` nodes (and then strips out the TERM, returning the definitionTerm node) or does not, acting then as a noop (returning undefined).

Reply:
It would be possible to write this operation as:
1. Lift `paragraph > text[value*=\n]` into [definitionTerm, paragraph]`
2. Lower `:not(definitionTerm)` into `definitionDescription` children

I think I prefer this but would be happy to rewrite if necessary :)